iceoryx_hoofs/vocabulary/include/iox/string.hpp

```diff
@@ -423,6 +424,13 @@ class string final
     template <typename T>
     IsStringOrCharArrayOrChar<T, bool> unsafe_append(const T& str) noexcept;
 
+    /// @brief direct access to the string's raw pointer. The access resizes the data correctly after the function call.
+    /// If the data written in the function exceeds the capacity, and the code does not segfault beforehand due to an
+    /// invalid access, a FATAL error occurs.
+    ///
+    /// @param [in] function composed by the raw data, current string size and the capacity
+    void unsafe_raw_access(const iox::function<void(char*, const uint64_t, const uint64_t)>& func) noexcept;
```

Comment:
Please use `iox::function_ref` instead of `ioc::function` when the function is not stored.

Oh, and could you define an alias fur the function and use that in `string.hpp` and here.